### ISSUE TITLE
Support multiple targets

### ISSUE THREAD

**Issue by mcandre** [rust-lang/cargo#3913]
Allow users to specify multiple targets, such as both x86_64-unknown-linux-gnu and x86_64-unknown-linux-musl, so that `cargo build` produces both artifacts.

By the way, how will the filename paths differentiate between multiple targets? Gox allows the target metadata to appear in the destination filepath, for example.

**Comment by alexcrichton** [comment]
Sounds like a reasonable feature to implement to me! I think the backend of Cargo isn't *quite* read to do this kind of operation, but with a small amount of refactoring it should be able to, namely [parameterizing `Kind`](https://github.com/rust-lang/cargo/blob/da4fadf6be3ef5030e411d3360244cf685617a58/src/cargo/ops/cargo_rustc/mod.rs#L38) to have more than one kind of `Target`.

**Comment by steveklabnik** [comment]
I'm hitting this today; I am writing a client and server both in wasm. Client crate needs to be compiled with one target, the server with another.

**Comment by michaeleiselsc** [comment]
if we did this, would it be possible to generate the IR only once but build for multiple backends?

**Comment by sfackler** [comment]
No, IR is platform dependent.

**Comment by dwijnand** [comment]
What does `cargo::ops::cargo_rustc::Kind` denote?

**Comment by ehuss** [comment]
> What does `cargo::ops::cargo_rustc::Kind` denote?

`Kind` was moved [here](https://github.com/rust-lang/cargo/blob/6d57b59ab505a803e4fb22f87b5ba60eb705163b/src/cargo/core/compiler/mod.rs#L45-L52). `Host` is used for proc-macros, plugins, and build scripts which must be compiled for the host architecture. `Target` is used when the user specifies `--target`.

**Comment by ehuss** [comment]
Closing in favor of tracking issue #8176.
